### PR TITLE
Fix build for macOS

### LINKED ISSUE #123: Publish latest release on Maven
Right now latest release [v2.0.1]() has failed on publishing to Maven (https://github.com/src-d/infrastructure/issues/1125).

The idea is, in order to unblock https://github.com/src-d/gitbase-spark-connector-enterprise/pull/93 to release a `v2.0.2-SNAPSHOT`.

The last attempt of releasing from master though has failed on macOS cross-compilation

TODOs:
 - [x] add new CI profile for macOS cross-compilation to `.travis.yml`
    In order to reproduce a failure from v2.0.2 release
   ```
    - <<: *release
      stage: test
      script: ./sbt assembly test
   ```
 - [x] fix macOS build that seems to be failing on STDLIB usage
    [Build logs](https://travis-ci.org/bblfsh/scala-client/jobs/577311092)
   ```
   
   In file included from src/main/native/org_bblfsh_client_v2_libuast_Libuast.cc:14:
   In file included from src/main/resources/libuast/libuast.hpp:6:
   In file included from /home/travis/osxcross/bin/../SDK/MacOSX10.11.sdk/usr/include/c++/v1/unordered_map:350:
   /home/travis/osxcross/bin/../SDK/MacOSX10.11.sdk/usr/include/c++/v1/__hash_table:1163:43: error: exception specification in declaration does not match previous declaration
   __hash_table<_Tp, _Hash, _Equal, _Alloc>::__hash_table()
                                             ^
   /home/travis/osxcross/bin/../SDK/MacOSX10.11.sdk/usr/include/c++/v1/__hash_table:850:5: note: previous declaration is here
       __hash_table()
       ^
   /home/travis/osxcross/bin/../SDK/MacOSX10.11.sdk/usr/include/c++/v1/__hash_table:1231:43: error: exception specification in declaration does not match previous declaration
   __hash_table<_Tp, _Hash, _Equal, _Alloc>::__hash_table(__hash_table&& __u)
                                             ^
   /home/travis/osxcross/bin/../SDK/MacOSX10.11.sdk/usr/include/c++/v1/__hash_table:864:5: note: previous declaration is here
       __hash_table(__hash_table&& __u)
       ^
   2 errors generated.
   ```
 - [x] release a `v2.0.2-SNAPSHOT` and make sure [it's published](https://oss.sonatype.org/content/repositories/snapshots/org/bblfsh/bblfsh-client/)

### PR DESCRIPTION
**Pull request by ncordon**
* Adds a job in travis to check release setting, i.e. would the project cross compile correctly in the release stage?
* Solves cross compilation issues for MacOS. Right now there is trouble cross compiling the project for sdk version 10.11.

Closes #123 

Signed-off-by: ncordon <nacho.cordon.castillo@gmail.com>

---
This change is [Reviewable](https://reviewable.io/reviews/bblfsh/scala-client/124)

### CODE REVIEW

**Comment by bzz** [comment]
Nice, we have a CI failure 💃

**Comment by bzz** [comment]
bollocks, we forgot `- cd $TRAVIS_BUILD_DIR` in the script section! 🤦‍♂

File: .travis.yml [hunk 1]
```diff
@@ -58,11 +57,15 @@ jobs:
...
         - ./sbt assembly || travis_terminate 1
         - ./sbt publishLocal
         - openssl aes-256-cbc -K $encrypted_97aef7f4ae04_key -iv $encrypted_97aef7f4ae04_iv -in key.asc.enc -out key.asc -d
         - gpg --no-default-keyring --primary-keyring ./project/.gnupg/pubring.gpg --secret-keyring ./project/.gnupg/secring.gpg --keyring ./project/.gnupg/pubring.gpg --fingerprint --import key.asc
         - ./sbt publishSigned
         - ./sbt sonatypeRelease
+    - <<: *release
+      name: 'Check that project can be cross-compiled correctly'
+      stage: test
+      script: ./sbt assembly test
```
Comment (bzz) [inline .travis.yml]:
```suggestion
      script:
        - ./sbt assembly test
        - cd $TRAVIS_BUILD_DIR
```

Comment (ncordon) [reply .travis.yml]: I have included `cd $TRAVIS_BUILD_DIR` at the end of `release install`, sinc it also was needed for `release script`. Now we have reproduced the same error as in the release. I am going to try and fix the MacOs sdk problem. Just to clarify, did you have to package manually the sdk and upload it to `https://s3.dockerproject.org/darwin/v2/MacOSX10.11.sdk.tar.xz`?

Comment (ncordon) [reply .travis.yml]: I have the 10.13 version here https://github.com/phracker/MacOSX-SDKs/releases

Comment (bzz) [reply .travis.yml]: > I have included cd $TRAVIS_BUILD_DIR at the end of release install

Is that somewhere in this PR?

**Comment by bzz** [comment]
Locally on linux, does the

```
        - curl -sSL "https://codeload.github.com/tpoechtrager/osxcross/tar.gz/${OSXCROSS_REV}" | tar -C "${PWD}" --strip=1 -xzf -
        - curl -sSL -o tarballs/MacOSX${SDK_VERSION}.sdk.tar.xz ${OSXCROSS_SDK_URL}
        - UNATTENDED=yes ./build.sh >/dev/null
```

work for you, with the the new archive?

**Comment by ncordon** [comment]
> Locally on linux, does the
> 
> ```
>         - curl -sSL "https://codeload.github.com/tpoechtrager/osxcross/tar.gz/${OSXCROSS_REV}" | tar -C "${PWD}" --strip=1 -xzf -
>         - curl -sSL -o tarballs/MacOSX${SDK_VERSION}.sdk.tar.xz ${OSXCROSS_SDK_URL}
>         - UNATTENDED=yes ./build.sh >/dev/null
> ```
> 
> work for you, with the the new archive?

What was not working for me was the line  `curl -sSL -o tarballs/MacOSX${SDK_VERSION}.sdk.tar.xz ${OSXCROSS_SDK_URL}` but because of permissions of `curl`. In travis it works. Locally what did worked for me was (and maybe it makes sense to do it on the travis also):

```
wget ${OSXCROSS_SDK_URL} -O tarballs/MacOSX${SDK_VERSION}.sdk.tar.xz
```

**Comment by ncordon** [comment]
I know this is ultra hacky. I have documented a proposal for improving this (and why it has to be so hacky right now) in this issue https://github.com/bblfsh/scala-client/issues/87

**Comment by ncordon** [comment]
___
*[build.sbt, line 78 at r4](https://reviewable.io/reviews/bblfsh/scala-client/124#-LnNwrz58mcUVtviBDqS:-LnNxtONBj2-CZwlo4Ym:b-o7hs8s) ([raw file](https://github.com/bblfsh/scala-client/blob/4add86111387fc136f7724247b19be30b0babd79/build.sbt#L78)):*
<details><summary><i>Previously, creachadair (M. J. Fromberger) wrote…</i></summary><blockquote>

Can these be `val const`?
</blockquote></details>

`val`s are immutable in Scala (`var`s are the ones that act as modifiable variables). It is a normal confusion though :)

**Comment by ncordon** [comment]
___
*[.travis.yml, line 23 at r4](https://reviewable.io/reviews/bblfsh/scala-client/124#-LnNvCXeB0IpfBBmb4dh:-LnNyFhB4Et5MA8bs1a4:b-w8lvyo) ([raw file](https://github.com/bblfsh/scala-client/blob/4add86111387fc136f7724247b19be30b0babd79/.travis.yml#L23)):*
<details><summary><i>Previously, creachadair (M. J. Fromberger) wrote…</i></summary><blockquote>

Maybe this should be done as a separate change, but it would be nice to have some documentation about what these values are for. Particularly in this case as we are adding a separate patch explicitly during the build process: When someone wants to update the base version of `osxcross`, for example, how can they safely do that without disrupting the patch? (Or: What do they have to do to update the patch at the same time? Or: Can we get rid of the patch once a certain release hits? Or whatever is true here).
</blockquote></details>

I will add a comment explaining it: both `OSXCROSS_PATCH_REV` and `OSXCROSS_REV` are the revisions of [osxcross](https://github.com/tpoechtrager/osxcross) repo. The `OSXCROSS_REV` was already used to retrieve a version of that tool that did not include the `.patch` file we need. I tried updating the former one too (because `OSXCROSS_PATCH_REV` is newer), but everything broke again, both locally and in Travis. To update `osxcross` one should update `OSXCROSS_REV`, the other one is just used to download a very specific `.patch` file. I would not recommend doing it for now though, because the `.patch` is applied by lines, and if the versions of the `stdlib` the in the repo changed, we would not be able to patch them. All of this boilerplate should go away if as I discussed [here](https://github.com/bblfsh/libuast/issues/111) we drop support for macOs 10.6 and bump to 10.7 everywhere, but I want feedback from mac users from the team / company.

**Comment by ncordon** [comment]
___
*[.travis.yml, line 65 at r4](https://reviewable.io/reviews/bblfsh/scala-client/124#-LnNvudh9vN2i87bYVlO:-LnO-pa3D3ivga4mh2aH:bgggigs) ([raw file](https://github.com/bblfsh/scala-client/blob/4add86111387fc136f7724247b19be30b0babd79/.travis.yml#L65)):*
<details><summary><i>Previously, creachadair (M. J. Fromberger) wrote…</i></summary><blockquote>

Will we eventually be able to get rid of this? Can you say something about what conditions would allow us to not need this anymore? If possible is there an issue link in the underlying project?
</blockquote></details>

Yes, use `OSX_VERSION_MIN=10.7` instead of `OSX_VERSION_MIN=10.6` should solve the problem and delete a lot of the hacky things we have here

**Comment by ncordon** [comment]
___
*[build.sbt, line 77 at r4](https://reviewable.io/reviews/bblfsh/scala-client/124#-LnNwF5pER52epe-GjOr:-LnO02EZCaOe-3oLdyIJ:b-6l9ru3) ([raw file](https://github.com/bblfsh/scala-client/blob/4add86111387fc136f7724247b19be30b0babd79/build.sbt#L77)):*
<details><summary><i>Previously, creachadair (M. J. Fromberger) wrote…</i></summary><blockquote>

I suggest not requiring extra whitespace around these values, but insert spaces when composing the command line as needed. Or if that is too big a pain, I think we should at least document that those spaces are needed so that future edits don't clobber them. (Also that quoting is required around internal whitespace, though in the arguments we have here there aren't any)
</blockquote></details>

Will change it : )

**Comment by ncordon** [comment]
___
*[.travis.yml, line 65 at r4](https://reviewable.io/reviews/bblfsh/scala-client/124#-LnNvudh9vN2i87bYVlO:-LnO3v8uEmJvnDSp6St5:blcqxjn) ([raw file](https://github.com/bblfsh/scala-client/blob/4add86111387fc136f7724247b19be30b0babd79/.travis.yml#L65)):*
<details><summary><i>Previously, creachadair (M. J. Fromberger) wrote…</i></summary><blockquote>

Is there any compelling reason for us _not_ to use 10.7 instead of 10.6? Snow Leopard was released in 2009, I think people have had long enough to upgrade. :grin:

I'm fine with doing this for now, though, to keep things working if you think that should be a separate question.
</blockquote></details>

Not to the best of my knowledge, I would root for upgrading to that version

**Pull request merged**